src/main/common/utils/numericUtils.ts

```diff
@@ -14,3 +14,37 @@ export function toNumberOrUndefined (value: any): number {
 
   return isNaN(numberVal) ? undefined : numberVal
 }
+
+export function containsAValidComma (input: string): boolean {
+  const output: string[] = input.split(',')
+
+  return output[output.length - 1].length >= 3
+}
+
+export function toAValidFloatOrOriginalValue (input: string): any {
```

Comment:
Looking at the method I would say it we do not have to distinguish numbers with fraction from numbers without it.

Reply:
Another thing is that this method is doing two things, it approaches deserialisation and if impossible it gives up by returning original. That is a bot too much IMO.

I propose the following approach:
1. deserialisation logic is responsible for deserialisation only and if impossible returns NaN or undefined
2. UI component is responsible for rendering correct value in the following way:
2.1. if value is valid it renders it as is using `valueFor`
2.2. if value is invalid the raw value is rendered from the form using `rawDataFor`

See: https://github.com/hmcts/cmc-citizen-frontend/pull/293/files#diff-8e8267522c3e829b11e0fcae1908cdb4R50 - implementation that stores raw form data in form object